mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -447,14 +346,17 @@ def wkf_cancel(self, cr, uid, ids, context=None):
     def wkf_close(self, cr, uid, ids, context=None):
         """Change state from in progress to closed"""
         o = self.browse(cr, uid, ids, context=context)[0]
-        done_states = ['done', 'cancelled']
+        # TODO make it more friendly
+        done_states = ['done', 'cancelled', 'settled', 'rejected']
         if (o.immediate_action_id
-                and o.immediate_action_id.state not in done_states):
+                and o.immediate_action_id.stage_id.name.lower()
```

Comment:
The object mgmtsystem.action doesn't have states anymore. I can add them back and ignore stages.